default.nix

```diff
@@ -1,6 +1,16 @@
 { pkgs ? import <nixpkgs> { } }:
 
-rec {
+let
+
+  flake = (import
+    (let lock = builtins.fromJSON (builtins.readFile ./flake.lock);
+    in fetchTarball {
+      url =
+        "https://github.com/edolstra/flake-compat/archive/${lock.nodes.flake-compat.locked.rev}.tar.gz";
+      sha256 = lock.nodes.flake-compat.locked.narHash;
+    }) { src = ./.; }).defaultNix;
```

Comment:
This feels pretty ugly to me. Why not simply change the `pkgs.fetchFromGitLab` to `fetchTarball` when fetching nmd and nmt?

Reply:
This is only a compatibility layer that should progressively become useless as we transition to flakes and the new nix CLI.

> Why not simply change the `pkgs.fetchFromGitLab` to `fetchTarball` when fetching nmd and nmt?

We could, but if we're going to use flakes, we might as well use them fully, i.e. for all inputs. I think they're a nicer way to manage dependencies, and they make updating and overriding easier.

Reply:
I agree that eventually it will be worth switching over to flakes but I'm skeptical about delving too deep while the feature is marked as experimental. At least unless there is a really compelling reason. But in this case there is a straight forward way to avoid the pkgs reference without touching flakes and needing compatibility shims.

I also would like to move the nmt and nmd projects over to sourcehut at some point and it doesn't seem like the flake-compat project supports the sourcehut URLs. I'll probably maintain a mirror at the gitlab location so this is not a blocker, though.

Reply:
It's also not a blocker because we can just use URLs like `git+https://git.sr.ht/~rycee/nmt`.

Reply:
> I'm skeptical about delving too deep while the feature is marked as experimental.

It is only marked as experimental to not have to create backwards compatibility while it still might change.